### DISCUSSION TITLE
Get API path parameter single get method

### DISCUSSION THREAD

**Discussion by shrikantnarayankar15** [fastapi/fastapi#6643] [Questions] [answered]
How to enable one single get method to work with below 2 cases
@app.get('/users/{id}')

1. If I use "/users" only then all the users should be listed
2. If I use "/users/1" only user with id 1 should be displayed.
Can this be achieved with one single get method?
@app.get('/users/{id}')

**Comment by flashgorman-chorus** [comment, marked as answer]
I'm no expert, but I don't think you can, and I'm not sure why you would want to.  If you're trying to consolidate code, you should just register the two different URIs, against two different functions, one that takes no args and one that takes a single `id` arg, and then have one delegate to the other, or have them both delegate to some 3rd, private/protected function:

```python
@app.get("/users")
def get_users():
    return _get_users()

@app.get("/users/{id}")
def get_user(id: str):
    return _get_users(id)[0]

def _get_users(id: str = None):
    # do whatever you want here...
```

**Comment by dstlny** [comment]
You could do the following:

```python

@app.get('/users')
def get_users(
    id: Optional[int] = None
):
    ...
```

This way, both the following URL's will work:
`/users` & `/users?id=12`

But as @flashgorman-chorus said, just split them into two endpoints w/ a common function call.

**Comment by shrikantnarayankar15** [comment]
Thanks.....I guess its impossible to write such method.

**Comment by tiangolo** [comment]
Thanks for the help here everyone! 👏  🙇 

Thanks for reporting back and closing the issue @shrikantnarayankar15  👍

I agree that it would be better to separate that logic into two functions.

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
